packages/react/src/elements/components/GlobalNav/TopNav/Notifications.js

```diff
@@ -18,7 +18,11 @@ class Notifications extends Component {
   };
 
   componentWillReceiveProps = nextProps => {
-    if (nextProps && nextProps.children.length > this.props.children.length) {
```

Comment:
Fixes a corner case where this component does not have current or incoming children.